src/jormungandr/jormungandr-lib/src/interfaces/updates.rs

```diff
@@ -0,0 +1,48 @@
+use crate::interfaces::{
+    config_params::FromConfigParamError, BlockDate, ConfigParams, ConsensusLeaderId,
+};
+use chain_impl_mockchain::{certificate::UpdateProposal, update::UpdateProposalState};
+use serde::{Deserialize, Serialize};
+
+#[derive(Deserialize, Serialize, Debug)]
+pub struct UpdateProposalStateDef {
+    pub proposal: UpdateProposalDef,
+    pub proposal_date: BlockDate,
+    pub votes: Vec<ConsensusLeaderId>,
+}
+
+impl TryFrom<UpdateProposalState> for UpdateProposalStateDef {
+    type Error = FromConfigParamError;
+    fn try_from(state: UpdateProposalState) -> Result<Self, FromConfigParamError> {
+        Ok(UpdateProposalStateDef {
+            proposal: state.proposal.try_into()?,
+            proposal_date: state.proposal_date.into(),
+            votes: state
+                .votes
+                .into_iter()
+                .map(|(x, _)| ConsensusLeaderId::from(x.clone()))
```

Comment:
Are you sure that it is needed `x.clone()` here, seems it should work just with `x`.

Reply:
unfortunately into_iter impl for Hamt return iter for inner collection